File: ASP .NET/Clients/Constants/ApiConstants.cs
namespace Clients.Constants;

/// <summary>
/// Constantes centralizadas para la API
/// Evita strings duplicados y facilita mantenimiento
/// Equivalente a ApiConstants.java en la API JAVA
/// </summary>
public static class ApiConstants
{
    // ==================== MENSAJES DE ÉXITO ====================
    public const string MessageLoginSuccess = "Login exitoso";
    public const string MessageRegisterSuccess = "Usuario registrado exitosamente. Revisa tu correo para verificar la cuenta.";
    public const string MessageProfileUpdated = "Perfil actualizado exitosamente";
    public const string MessagePasswordUpdated = "Contraseña actualizada exitosamente";
    public const string MessageProfilePictureUpdated = "Foto de perfil actualizada exitosamente";
    public const string MessageProfileDeleted = "Perfil eliminado exitosamente";
    public const string MessageEmailVerified = "Email verificado con éxito";
    public const string MessageTokenRefreshed = "Token refrescado exitosamente";
    public const string MessageProfileFetched = "Perfil obtenido exitosamente";
    public const string MessageUsersFetched = "Lista de usuarios obtenida exitosamente";

    // ==================== MENSAJES DE ERROR ====================
    public const string ErrorInvalidCredentials = "Email o contraseña incorrectos";
    public const string ErrorUnauthorized = "No autenticado";
    public const string ErrorForbidden = "Acceso denegado";
    public const string ErrorUserNotFound = "Usuario no encontrado";
    public const string ErrorUserNotActive = "Usuario no verificado o inactivo";
    public const string ErrorEmailExists = "El email ya está registrado";
    public const string ErrorNickExists = "El nick ya está en uso";
    public const string ErrorInvalidToken = "Token inválido o expirado";
    public const string ErrorVerificationFailed = "Verificación fallida";
    public const string ErrorNoImage = "Por favor selecciona una imagen";
    public const string ErrorImageTypeInvalid = "Tipo de archivo no permitido. Solo se permiten imágenes (jpg, png, gif, webp)";
    public const string ErrorImageSaveFailed = "Error al guardar imagen";
    public const string ErrorImageDeleteFailed = "Error al eliminar imagen";
    public const string ErrorInternalError = "Error interno del servidor";
    public const string ErrorValidationError = "Error de validación";

    // ==================== RUTAS Y PATHS ====================
    public const string ApiPrefix = "/api";
    public const string AuthEndpoint = ApiPrefix + "/auth";
    public const string UserEndpoint = ApiPrefix + "/user";
    public const string ProfileEndpoint = ApiPrefix + "/profile";
    public const string ImagesEndpoint = ApiPrefix + "/images";

    // ==================== CONFIGURACIÓN JWT ====================
    public const string JwtCookieName = "auth_token";
    public const string JwtCookiePath = "/";
    public const int JwtCookieMaxAge = 86400; // 24 horas
    public const string JwtSameSite = "Lax";

    // ==================== CONFIGURACIÓN ROLES ====================
    public const string RolePrefix = "ROLE_";
    public const string RoleAdmin = "ADMIN";
    public const string RoleUser = "USER";
    public const string RolePremium = "PREMIUM";

    // ==================== REGEXP Y VALIDACIONES ====================
    public const string FilenamePattern = @"^[a-zA-Z0-9._-]+$";
    public const string ImagePathPattern = @"^[a-zA-Z0-9/_.-]+$";

    // ==================== ERRORES HTTP ====================
    public const string ErrorMethodNotAllowed = "Método HTTP no permitido";
    public const string ErrorNotFound = "Recurso no encontrado";
}

File: JAVA/Clients/clients/src/main/java/com/futureprograms/clients/util/ApiConstants.java
package com.futureprograms.clients.util;

/**
 * Constantes centralizadas para la API
 * Evita strings duplicados y facilita mantenimiento
 */
public final class ApiConstants {

    // ==================== MENSAJES DE ÉXITO ====================
    public static final String MSG_LOGIN_SUCCESS = "Login exitoso";
    public static final String MSG_REGISTER_SUCCESS = "Usuario registrado exitosamente. Revisa tu correo para verificar la cuenta.";
    public static final String MSG_PROFILE_UPDATED = "Perfil actualizado exitosamente";
    public static final String MSG_PASSWORD_UPDATED = "Contraseña actualizada exitosamente";
    public static final String MSG_PROFILE_PICTURE_UPDATED = "Foto de perfil actualizada exitosamente";
    public static final String MSG_PROFILE_DELETED = "Perfil eliminado exitosamente";
    public static final String MSG_EMAIL_VERIFIED = "Email verificado con éxito";
    public static final String MSG_TOKEN_REFRESHED = "Token refrescado exitosamente";
    public static final String MSG_PROFILE_FETCHED = "Perfil obtenido exitosamente";
    public static final String MSG_USERS_FETCHED = "Lista de usuarios obtenida exitosamente";

    // ==================== MENSAJES DE ERROR ====================
    public static final String ERR_INVALID_CREDENTIALS = "Email o contraseña incorrectos";
    public static final String ERR_UNAUTHORIZED = "No autenticado";
    public static final String ERR_FORBIDDEN = "Acceso denegado";
    public static final String ERR_USER_NOT_FOUND = "Usuario no encontrado";
    public static final String ERR_USER_NOT_ACTIVE = "Usuario no verificado o inactivo";
    public static final String ERR_EMAIL_EXISTS = "El email ya está registrado";
    public static final String ERR_NICK_EXISTS = "El nick ya está en uso";
    public static final String ERR_INVALID_TOKEN = "Token inválido o expirado";
    public static final String ERR_VERIFICATION_FAILED = "Verificación fallida";
    public static final String ERR_NO_IMAGE = "Por favor selecciona una imagen";
    public static final String ERR_IMAGE_TYPE_INVALID = "Tipo de archivo no permitido. Solo se permiten imágenes (jpg, png, gif, webp)";
    public static final String ERR_IMAGE_SAVE_FAILED = "Error al guardar imagen";
    public static final String ERR_IMAGE_DELETE_FAILED = "Error al eliminar imagen";
    public static final String ERR_INTERNAL_ERROR = "Error interno del servidor";
    public static final String ERR_VALIDATION_ERROR = "Error de validación";

    // ==================== RUTAS Y PATHS ====================
    public static final String API_PREFIX = "/api";
    public static final String AUTH_ENDPOINT = API_PREFIX + "/auth";
    public static final String USER_ENDPOINT = API_PREFIX + "/user";
    public static final String PROFILE_ENDPOINT = API_PREFIX + "/profile";
    public static final String IMAGES_ENDPOINT = API_PREFIX + "/images";

    // ==================== CONFIGURACIÓN JWT ====================
    public static final String JWT_COOKIE_NAME = "auth_token";
    public static final String JWT_COOKIE_PATH = "/";
    public static final int JWT_COOKIE_MAX_AGE = 86400; // 24 horas
    public static final String JWT_SAME_SITE = "Lax";

    // ==================== CONFIGURACIÓN ROLES ====================
    public static final String ROLE_PREFIX = "ROLE_";
    public static final String ROLE_ADMIN = "ADMIN";
    public static final String ROLE_USER = "USER";
    public static final String ROLE_PREMIUM = "PREMIUM";

    // ==================== REGEXP Y VALIDACIONES ====================
    public static final String FILENAME_PATTERN = "^[a-zA-Z0-9._-]+$";
    public static final String IMAGE_PATH_PATTERN = "^[a-zA-Z0-9/_.-]+$";

    // ==================== ERRORES HTTP ====================
    public static final String ERR_METHOD_NOT_ALLOWED = "Método HTTP no permitido";
    public static final String ERR_NOT_FOUND = "Recurso no encontrado";

    private ApiConstants() {
        throw new AssertionError("No se puede instanciar esta clase");
    }
}
